pyasic/miners/device/firmware.py

```diff
@@ -21,26 +21,47 @@
 class StockFirmware(BaseMiner):
     firmware = MinerFirmware.STOCK
 
+    async def upgrade_firmware(self, *, url: str = None, version: str = "latest") -> bool:
```

Comment:
No need to define any of these yet.  They are automatically called into super(), so this is redundant.

Reply:
The way this is done here is correct - https://github.com/UpstreamData/pyasic/pull/157/commits/eefb055a3f2c67f9eecd250a288a63a1dee31142